toc.md

```diff
@@ -165,34 +190,50 @@ module Workerholic
 end
 ```
 
-Our internal `kill` method simply sets the `alive` instance variable from `true` to `false`. Afterwards, we join each of our worker threads with the main thread. This way, we ensure that the workers can have a chance to finish their final jobs before actually shutting down.
+This will allow the worker to finish executing its current job. Afterwards, we `join` each of our worker with our main thread of execution. We will touch on why this is important in a later section, for now it is only useful to know that the workers will be waited on before Workerholic exits.
 
 ### Efficiency
-Let's shift gears a little bit and start talking about how Workerholic is efficient.
+
+Efficiency is another core feature for a BJP. Why does a BJP need to be efficient? How can we build an efficient? Those are the questions that we will answer in the following sections.
 
 #### An Example Scenario
 
 ##### Scenario
-Suppose we have a huge Rails application with the follow statistics:
+
+Suppose we have a significantly big Rails application with the following statistics:
```

Comment:
"Suppose we have a relatively large production Rails application with the following constraints:"